vm/src/exceptions.rs

```diff
@@ -1297,9 +1335,19 @@ pub(super) mod types {
     fn os_error_new(cls: PyTypeRef, args: FuncArgs, vm: &VirtualMachine) -> PyResult {
         PyBaseException::slot_new(cls, args, vm)
     }
+    fn os_error_init(zelf: PyObjectRef, args: FuncArgs, vm: &VirtualMachine) -> PyResult<()> {
+        if args.args.len() >= 3 {
+            zelf.set_attr("filename", args.args[2].clone(), vm)?;
+            if args.args.len() >= 4 {
+                zelf.set_attr("filename2", args.args[args.args.len() - 1].clone(), vm)?;
+            }
+        }
 
-    fn base_exception_init(zelf: PyObjectRef, args: FuncArgs, vm: &VirtualMachine) -> PyResult<()> {
-        PyBaseException::init(zelf, args, vm)
+        let mut new_args = args;
+        if new_args.args.len() >= 2 {
+            new_args.args.truncate(2);
+        }
```

Comment:
This is a question because I don't know the spec very well. Exceptions usually takes infinite args and store it. What happens for more args than 4 in this case?

Reply:
It seems to work that no matter how we put arguments to its parenthesis (`work` means it does not throw exceptions).
But according to code and behavior. it works **as expected** when only the given argument length is between 2 and 5.
```python
>>> OSError(1,2).errno
1
>>> OSError(1,2).strerror
2
>>> OSError(1)
OSError(1)
>>> OSError(1).errno
```
https://github.com/python/cpython/blob/3.10/Objects/exceptions.c#L820-L876

Reply:
Let's discuss this point in #4194 ...